main.go

```diff
@@ -44,6 +45,14 @@ func main() {
 		"BalanceWei": balance,
 	}).Info("Pool Address Balance")
 
+	//Ensure that configured collateral in oracle matches contract collateral
+	contractCollateral, err := onchain.GetContractCollateral(retry.Attempts(uint(cfg.NumRetries)))
```

Comment:
can you have a look into `GetRetryOpts` ? this allows to have a default value the retry logic.

so the idea is to have just `GetContractCollateral()` without any parameter as input.

then when calling `GetRetryOpts` without providing any `opts`, default values will be used.

if anyone then want to use `GetContractCollateral` with a custom amount of attempts, the can provide the argument they desire.

so **tldr**: lets just have `GetContractCollateral()` that under the hood uses the default value.